### ISSUE TITLE
@GraphQLApi is not a bean defining annotation, yet TCK does not mark classes as beans

### ISSUE THREAD

**Issue by tomas-langer** [eclipse/microprofile-graphql#291]
According to spec, the following should be true:
```
GraphQL endpoints must be annotated with the @GraphQLApi annotation.
The @GraphQLApi annotation is a marker annotation that identifies a CDI Bean as a GraphQL
Endpoint
```

It identifies a CDI bean as a GraphQL - yet classes in TCK, such as `HeroFinder` are not annotated with a bean defining annotation, so these are not provided as beans in CDI extensions

Do I understand correctly, that the `HeroFinder` should be annotated with `@ApplicationScoped`, or should we treat `@GraphQLApi` as a bean defining annotation (or did I misunderstand this altogether)?

Thanks

**Comment by phillip-kruger** [comment]
As far as I know (or remember) we said that the default will be `@ApplicationScope`, so if a bean is only annotated with `GraphQLApi` we use application scope. Else the use can optionally add the scope, then we use that. 
@andymc12 - did I remember correct ?

**Comment by tomas-langer** [comment]
According to this blog (and a CDI book we have in paper): https://blogs.oracle.com/theaquarium/default-cdi-enablement-in-java-ee-7#:~:text=By%20default%2C%20if%20you%20specify,not%20'all'%20is%20assumed
`bean-discovery-mode` default value is `annotated`, not `all`.

So the bean cannot be seen unless annotated with a bean defining annotation.

**Comment by phillip-kruger** [comment]
The extension makes it a bean. See 16.7. Registering a Bean here https://docs.jboss.org/weld/reference/latest/en-US/html/extend.html

**Comment by tomas-langer** [comment]
That is not what the Graph-QL specification says:
`The @GraphQLApi annotation is a marker annotation that identifies a CDI Bean as a GraphQL Endpoint`

This says that this annotation identifies an existing CDI bean as a GraphQL Endpoint, not that it marks the class as a bean and a GraphQL Endpoint.

Maybe this is a wording issue, but it a very important one.

**Comment by tomas-langer** [comment]
Also the `Registering a Bean` adds a class as a bean, it does not add all annotated types.

**Comment by phillip-kruger** [comment]
Agree, let's make it more clear in the spec. I'll add it to the meeting minutes for today. You are welcome to join :)

**Comment by tomas-langer** [comment]
I will try to explain why this is a problem - as far as I know, CDI SPIs for extension do not allow adding a new bean defining annotations - so there is no API that would allow me to tell CDI implementation "please mark all classes annotated with this annotation as beans with default scope set to `@ApplicationScoped`"
As a result, I need to do one of the following:

1. Use implementation specific approach to add a bean defining annotation
2. Use Jandex (again implementation specific) to discover annotated classes
3. Use custom classpath scanning tool to discover annotated classes

I see a problem with relation to `beans.xml` - CDI understands it and correctly processes classes in bean archives accordingly. Custom handling may introduces problems.

Another issue I see here is inconsistency between MP specification - for example in Health checks, the health check class must be a CDI bean (and we pass TCK without the need to process all classes on classpath).

**Comment by phillip-kruger** [comment]
If we add `@ApplicationScope` would a user still be able mark it as another scope?
